ArtOfIllusion/src/artofillusion/MaterialPreviewer.java

```diff
@@ -380,9 +381,9 @@ private void mouseDragged(MouseDraggedEvent e)
       return;
     Graphics g = getComponent().getGraphics();
     Point dragPoint = e.getPoint();
-    if (e.isMetaDown())
+    if (mouseButtonThree(e))
     {
-      if (e.isControlDown())
+      if (mouseButtonTwo(e))
```

Comment:
Two mousebuttons.
```suggestion
       if (e.isControlDown())
```

Ctrl should turn the move to zoom. Something does happen with two mouse buttons down, but that was not a very well controlled event.

There also seems to be other clitches with the view manipulation here, but that'd take a different study and different update.

Reply:
There seems to be more to this than I (or I suppose anybody) would have expected.

With `mouseButtonThree(e)` in the outer if-line of `mouseReleased` the drag ghost is drawn as you would expect: Right button+ctrl zoom the ghost during drag but at `mouseReleased` the rendered image is not zoomed but rotated!

It with control down the `mouseButtonThree(e)` returns false in the case, that the event is `MouseReleased` but with `MouseDragged`event it seems to work correctly. If I change mb3 back to isMetaDown for `mouseReleased`, then the rendered image appears zoomed as it should.

With control-key up the moving of the view with the right buttpn works as expected, first dragging the ghost and in at mouse-up rendering the image where the ghost was.

This is not the the kind of news anybody likes at this point but, if I'm not badly mistaken, this could be causing similar problems elsewhere as well...

Reply:
I have no idea - My first two thoughts did not pan out. That's frustrating.

Reply:
@peastman I've hit a snag here. I did a test, and bare AWT events do track modifiers correctly in `mouseRelease,` even when run through the `mouseButtonThree()` method. That suggests that either:

- The event is intercepted and changed by something else in AOI
- Buoy handles these a little differently? The wrapper seems thin, though. I don't know what it might be doing differently.

Any thoughts on where to begin looking?